papermill/execute.py

```diff
@@ -13,26 +15,75 @@
 from six import string_types
 
 
-class PapermillExecutePreprocessor(ExecutePreprocessor):
+def preprocess(self, nb, resources):
+    """
+    This function is for monkey patching the nbconvert Preprocessor.preprocess method.
+    We are doing this for the following reasons:
+    1. Notebooks will stop executing when they encounter a failure but not raise a CellException.
+       This allows us to save the notebook with the traceback even though a CellExecutionError
+       was encountered.
+
+    2. We want to write the notebook as cells are executed. We inject our logic for that here.
 
-    def preprocess(self, nb, resources):
-        """
-        The default preprocess behavior if allow_errors = True is to continue executing all the cells.
-        We want the notebook to cease execution and then write out the state of the notebook with the traceback
-        on the cell that failed.
-        """
-        # Clear out all outputs prior to execution.
-        for cell in nb.cells:
-            cell.outputs = []
+    3. We will be included cell execution metrics.
 
-        # Catch CellExecutionError if thrown which lets this method finish so we can write the state of the notebook
-        # to disk.
-        try:
-            nb, resources = super(PapermillExecutePreprocessor, self).preprocess(nb, resources)
-        except CellExecutionError:
-            pass
+    Original Docstring:
```

Comment:
I'd either cut out this line or make this all fit together in one narrative for the docstring (can be broken up in paragraphs as needed).